test/graph_rewrites/test_ng_var_update_ng_tensor.cc

```diff
@@ -40,85 +41,125 @@ namespace tensorflow {
 namespace ngraph_bridge {
 namespace testing {
 
-class NGVarUpdateNGTensorOpTest : public tensorflow::OpsTestBase {};
+TEST(NGVarUpdateNGTensorOpTest, SimpleGraph1) {
+  Graph g(OpRegistry::Global());
+  PartialTensorShape varShape({2, 2});
+
+  Node* var_node;
+  ASSERT_OK(NodeBuilder("var_node", "NGraphVariable")
+                .Attr("shape", varShape)
+                .Attr("dtype", DT_FLOAT)
+                .Attr("just_looking", false)
+                .Attr("copy_to_tf", false)
+                .Attr("container", "")
+                .Attr("shared_name", "node1")
+                .Attr("ngraph_graph_id", 1)
+                .Attr("_ngraph_backend", "CPU")
+                .Finalize(&g, &var_node));
+
+  std::vector<DataType> input_types;
+  input_types.push_back(DT_FLOAT);
+  std::vector<DataType> output_types;
+  output_types.push_back(DT_FLOAT);
+  std::vector<NodeBuilder::NodeOut> inputs;
+  inputs.push_back(NodeBuilder::NodeOut(var_node, 0));
+  Node* encap_node;
+  ASSERT_OK(NodeBuilder("encap_node", "NGraphEncapsulate")
+                .Attr("Targuments", input_types)
+                .Attr("Tresults", output_types)
+                .Attr("ngraph_cluster", 1)
+                .Attr("ngraph_graph_id", 1)
+                .Attr("ngraph_backend", "CPU")
+                .Attr("ngraph_device_id", "1")
+                .Input(inputs)
+                .Finalize(&g, &encap_node));
+
+  NodeBuilder::NodeOut input_val = NodeBuilder::NodeOut(encap_node, 0);
+  Node* assign;
+  ASSERT_OK(NodeBuilder("assign", "Assign")
+                .Input(var_node)
+                .Input(input_val)
+                .Attr("T", DT_FLOAT)
+                .Finalize(&g, &assign));
+
+  Node* source = g.source_node();
+  Node* sink = g.sink_node();
+  g.AddEdge(source, Graph::kControlSlot, var_node, Graph::kControlSlot);
+  g.AddEdge(assign, Graph::kControlSlot, sink, Graph::kControlSlot);
+
+  ASSERT_OK(RewriteForTracking(&g, 0));
+
+  map<string, Node*> node_map;
+  for (auto node : g.op_nodes()) {
+    node_map[node->name()] = node;
+  }
+  ASSERT_EQ(node_map.find("var_node_sync_node")->second->type_string(),
```

Comment:
You can assert on the right connections of the edges as that is where complexity lies.

Reply:
good point